### PR TITLE
test(browser): add browser sign out test cases

### PR DESCRIPTION
**Pull request by demonzoo**
## Summary

Add unit test cases for sign out in browser SDK

## Linear Issue Reference

[LOG-1571](https://linear.app/silverhand/issue/LOG-1571/add-unit-test-cases-for-browser-sign-out)

## Testing

Passed all test cases

### CODE REVIEW

File: packages/browser/src/index.test.ts [hunk 1]
```diff
@@ -97,4 +102,36 @@ describe('LogtoClient', () => {
...
+
+  describe('signOut', () => {
+    beforeEach(() => {
+      localStorage.setItem(idTokenStorageKey, 'id_token_value');
+      localStorage.setItem(refreshTokenStorageKey, 'refresh_token_value');
+    });
+
+    test('should call token revocation endpoint with requester', async () => {
+      const logtoClient = new LogtoClient({ endpoint, clientId, requester });
+      await logtoClient.signOut(postSignOutRedirectUri);
```
Comment (gao-sun) [inline packages/browser/src/index.test.ts:114]: will this send a real request?

Comment (gao-sun) [reply packages/browser/src/index.test.ts:114]: also should cover request failed?

Comment (demonzoo) [reply packages/browser/src/index.test.ts:114]: > will this send a real request?

No, the `requester` function is mocked at the very beginning of this test suite.

> also should cover request failed?

I'm afraid not, as we do not want to block sign-out flow even if the revocation is failed, so there is zero handling in the `try-catch` block of the `revoke` request.

Comment (gao-sun) [reply packages/browser/src/index.test.ts:114]: yes, i mean we should cover "even if the request failed but sign out is still non-blocking" case

Comment (demonzoo) [reply packages/browser/src/index.test.ts:114]: OK, will add that part shortly

**Review by gao-sun — APPROVED** [review]

**Pull request merged**